### ISSUE TITLE
No Usage Examples

### ISSUE THREAD

**Issue by Californian** [CloverHealth/pytest-pgsql#7]
There are no complete usage examples here. How do I use this, e.g., with sqlalchemy and alembic? Can models automatically be registered, or do they need to be individually created in tests?

**Comment by dargueta** [comment]
Can you give us an example of some code you'd like to test, and maybe we can work with that?

**Comment by dantimofte** [comment]
I'm also interested in examples. 

Could you write some test for this project : https://github.com/biofreack/data_scrapper 

Alembic is in https://github.com/biofreack/crypto_database

I think the project is simple enough and writing the tests shouldn't take too long if know how .

**Comment by dargueta** [comment]
> Could you write some test for this project : https://github.com/biofreack/data_scrapper

From what I can tell, that project is a webpage scraper and uses the database only for dumping data it's pulled from coinmarketcap.com. It doesn't do anything database-related aside from storage.

**Comment by dantimofte** [comment]
Yes, it's just storing some data but it uses alembic and has 2 functions in __main__.py for storing. When you for a more general example you mean with more database usage than inserts ?
 
I have a similar project that only stores data and uses the id of record for some additional processing.

I looked at this library and others as well and i don't understand how to write the tests. Maybe you can help me ?

**Comment by dantimofte** [comment]
> > Could you write some test for this project : https://github.com/biofreack/data_scrapper
> 
> From what I can tell, that project is a webpage scraper and uses the database only for dumping data it's pulled from coinmarketcap.com. It doesn't do anything database-related aside from storage.
> 
> That's also really project-specific. I was hoping for a more general example.

i wanted something like this : 
[](https://gist.github.com/dantimofte/0d85a6e7f548bf6cf8d68d7a3cea62fe)

**Comment by michaelfresco** [comment]
The book Essential SQLAlchemy has a really nice example with pytest.

https://github.com/oreillymedia/essential-sqlalchemy-2e/tree/master/ch04
